src/gwt/src/org/rstudio/studio/client/workbench/views/environment/dataimport/DataImportOptionsUiCsv.java

```diff
@@ -175,14 +185,84 @@ public void onValueChange(ValueChangeEvent<String> arg0)
       
       ChangeHandler changeHandler = new ChangeHandler()
       {
-         
          @Override
          public void onChange(ChangeEvent arg0)
          {
             updateEnabled();
             triggerChange();
          }
       };
+
+      ChangeHandler delimChangeHandler = new ChangeHandler()
+      {
+         @Override
+         public void onChange(ChangeEvent arg0)
+         {
+            if (delimiterListBox_.getSelectedValue() == "other")
+            {
+               globalDisplay_.promptForTextWithOption(
+                  "Other Delimiter",
+                  "Please enter a single character delimiter.",
+                  "",
+                  false,
+                  "",
+                  false,
+                  new ProgressOperationWithInput<PromptWithOptionResult>()
+                  {
+                     private void dismissAndUpdate(ProgressIndicator indicator, int newSelectIndex)
+                     {
+                        lastDelimiterListBoxIndex_ = newSelectIndex;
+                        delimiterListBox_.setSelectedIndex(newSelectIndex);
+
+                        indicator.onCompleted();
+
+                        updateEnabled();
+                        triggerChange();  
+                     }
+
+                     @Override
+                     public void execute(PromptWithOptionResult result,
+                                         ProgressIndicator indicator)
+                     {
+                        String otherDelimiter = result.input;
+                        
+                        if (otherDelimiter.length() != 1) {
+                           globalDisplay_.showErrorMessage("Incorrect Delimiter", "The specified delimiter is not valid.");
```

Comment:
Nit: Can this error be more specific? e.g. `The specified delimiter is not valid. Delimiters must consist of a single character.`